! # Tutorial Video Walkthrough
! # video_walkthrough_ai_chatgpt_usages.diff

This file keep track of all the infos given in the videos....

- TITLE
+ Using ChatGPT on a daily work as a P.O, Developer or for Q/A or Support and checking plagiarism if needed with Python
--- See more on the post at https://wp.me/p3Vuhl-3dF

- URLs
+ ARTICLE :: https://wp.me/p3Vuhl-3dF
+ CODE :: https://bit.ly/3YI5Ubl
+ WEBSITE ::  https://flaven.fr/

- QUICK DESCRIPTION

--- OBJECTIVE :: How to use ChatGPT to replace tedious work by AI + checking plagiarism if needed with Python

--- OUPUT :: Numerous example from Jira ticket for P.O, Q/A testing script in JS, WP plugin in PHP, Pyhton stuff, SQL code...

--- AUDIENCE :: Anyone who want to use ChatGPT on a daily basis.


- TEK GUIDELINES
--- ChatGPT :: https://chat.openai.com
--- Streamlit :: https://streamlit.io/
--- scikit-learn :: hhttps://scikit-learn.org/stable/index.html




- PATH CONSOLE
cd /Users/brunoflaven/Documents/03_git/BlogArticlesExamples/ai_chatgpt_usages/


- CONDA ENVIRONMENT
--- requirement using conda to manage development environment
--- create requirements file for the apps
pip freeze > requirements_plagiarism_checker_python.txt
pip install -r requirements_plagiarism_checker_python.txt


! CONTENT

+ 1 + TWO REMARKS ON CHATGPT
First, asking Chat GPT about moral issues is pointless and sometimes ludicrous!

Second, still about the "corpus", as we do not know exactly the corpus of texts on which ChatGPT was trained to perfect its learning, it is certainly obvious that all the documentation relating to the development languages has been included (SQL, PHP, Python, JavaScript and so on... etc.) but that they do not necessarily integrate the latest developments.


+ 2 + HOW TO DETECT PLAGIARISM?
The idea naturally came because of using GPT chat https://chat.openai.com/ is how do we ensure that is has been made by a real person or by an IA?
Consequently, the massive use of GPT Chat or any other form of AI device necessarily brings its corollary of control which comes down to the following question: Is it a human who wrote all or part of the text or code that I have under the eyes?

+ 3 + EXAMPLES GENERATED WITH CHATGPT FOR PRODUCT OWNER, DEVELOPER...

- 1. PO: generate Jira tickets for user stories, tasks, and bugs

! # Write a Jira ticket to create a button that automatically add the author tag to the editing content. The ticket must give a precise DOR (definition of ready) and DOD (definition of done) to clarify the Acceptance Criteria.

# 004_generate_code_ia_jira_ticket_add_tag_me_button.txt
# source for GPT query
# Write a Jira ticket to create a button that add automatically the author tag to the editing content. The ticket must give a precise DOR (definition of ready) and DOD (definition of done) to clarify the Acceptance Criteria.


Title: Create "Add Author Tag" Button

Description:
As a user, I want a button that adds the author tag to the content I am editing, so that it is clear who wrote the content.

DOR (Definition of Ready):

The scope of the feature is well defined and agreed upon
The acceptance criteria are clearly defined and agreed upon
The design of the button has been approved
The feature has been estimated and added to the sprint backlog
Any dependencies or potential blockers have been identified and addressed
Acceptance Criteria:

A "Add Author Tag" button is added to the content editor
Clicking the button will insert the author tag into the content at the current cursor position
The author tag includes the name of the user who is currently logged in
The button is clearly visible and intuitive to use
The feature is compatible with all supported browsers
DOD (Definition of Done):

The feature is fully implemented and tested
The code has been reviewed and meets our coding standards
Automated tests for the feature are written and passing
The feature is deployed to the testing environment and passes all acceptance tests
The feature is deployed to the production environment and is being used by users
Documentation for the feature is up-to-date and accurate

- 2. Q/A: Some testing examples

! Write the Cypress code to test that the fields in the post content: title, body, tags has been properly filled by the user and then save the post content created as draft

# 005_generate_code_ia_cypress_check_post_draft.js

- 3. DEV: Some Coding examples
--- 1 – Coding: Query example for GPT to generate code for MySQL
! WordPress custom SQL query examples

! # Explain with simple words the following code
global $wpdb;
$wpdb->insert(
    $wpdb->foods,
    array(
        'fruit' => 'apple',
        'year' => 2012
    ),
    array(
        '%s',
        '%d'
    )
);

This code uses the WordPress database object, $wpdb, to insert a new record into a table called foods. The new record contains two pieces of data: the name of a fruit, which is "apple", and a year, which is 2012. The array() function is used to create an array of values to be inserted into the table. The second array() function specifies the format of the data being inserted, with %s indicating a string and %d indicating a number.

The global keyword is used to make the $wpdb variable accessible within the function or scope it is being used in. This is necessary because $wpdb is defined outside of the function in the global scope.

In summary, this code inserts a new record into a WordPress database table called foods with the values "apple" and 2012 for the fruit name and year, respectively.

# 002_source_MySQL_GPT_query_explained_wpdb_insert.sql


! # Create table https://github.com/bflaven/BlogArticlesExamples/blob/e0d7c9858285eb547730db268ea0114ef8d5d3a2/extending_streamlit_usage/009_countries_crud_streamlit_app/sqlite_try_3_create_table.py

# tld : Country code top-level domain

! # In SQLite, write the code to create a table named countries with the following fields: name, tld, email, cca2, capital, callingCode. The user id must be a key and incremental.

sql_create_countries_table = """CREATE TABLE IF NOT EXISTS countries (
                                    id INTEGER PRIMARY KEY,
                                    name TEXT NOT NULL,
                                    tld TEXT NOT NULL UNIQUE,
                                    cca2 TEXT NOT NULL UNIQUE,
                                    capital TEXT NOT NULL UNIQUE,
                                    callingCode INTEGER,
                                    FOREIGN KEY (id) REFERENCES countries (id)
                                );"""

# 003_source_MySQL_GPT_sqllite_create_countries_table.sql

Here's the SQLite code to create a table named "countries" with the specified fields:



CREATE TABLE countries (
  id INTEGER PRIMARY KEY AUTOINCREMENT,
  name TEXT,
  tld TEXT,
  email TEXT,
  cca2 TEXT,
  capital TEXT,
  callingCode TEXT
);


In this code, the "id" field is defined as the primary key with the "INTEGER" data type, and it is set to auto-increment using the "AUTOINCREMENT" keyword. The remaining fields are defined with their respective data types of "TEXT".



--- 2 – Coding: Query example for GPT to generate code in Python
! Write python code with comments using beautyfulsoup that read a file named example_3.html and find out the H5 tag and H6 tag in the HTML source. The H5 will then a variable called "smallBookTitles", it has to be injected into a loop that create a list for all the books showed in the page example_3.html

! # Model: https://github.com/bflaven/BlogArticlesExamples/blob/34fe0f51a53b0ae38af614e66b9eeaf16facf97d/webscraping_with_python/slurpThatSoup_6.py

# 007_generate_code_python_slurpThatSoup.py

--- 3 – Coding: Query example for GPT to generate code in PHP for a plugin WordPress named “MySocialBuddy”

! Write a WordPress plugin that enable to manage different languages with flags

! # Write in PHP, a Wordpress plugin with comments, named MyFlagsSiteVersion. This Plugin MyFlagsSiteVersion will add a new entry in the main administration menu in "Settings". The name of the entrance will be MyFlagsSiteVersion. The purpose of this plugin is to manipulate a single custom fields for 2 languages (spanish, russian):  myflagssiteversion_sp, myflagssiteversion_ru. You can you write also tips for this field and general explanation at the beginning of the plugin page settings to explain how does the plugin work.

! # Model: https://github.com/bflaven/PluginWordpressForFun/blob/master/johann_flags_header/johann_flags_header.php


# FIRST CODE GENERATION
# 005_generate_php_plugin_wordpress_flags_header.php

# SECOND CODE GENERATION
# 006_generate_php_plugin_wordpress_flags_header.php



- 4 – Coding: Other attempts...


+ 4. DEV: ERRORS EXPLAINED

--- Some errors searchings quicker to look in stackoverflow.com. The error can be explained by ChatGPT, it is complementary source of information with stackoverflow.com for instance

! # sample for errors...
--- Explain the python error "TypeError: Protocols cannot be instantiated"
--- Explain the python error "TypeError: Protocols cannot be instantiated for Streamlit what is the solution?"
--- How to update python in anaconda (conda)?


+ 5. SUPPORT: MAILS FOR SUPPORT
--- 1 - Support: Query example for GPT to generate support mail for 2FA authentication
--- 2 - Support: Query example for GPT to generate support mail for user to explain how-to to login/logout and start private session
--- 3 - Support: Query example for GPT to generate support mail

--- 4 - Support: Query example for GPT to generate support mail

# 004_suppport_mail_site_down_emergency_hacked_please_stay_calm.txt
# source for GPT query
# write a support mail to users to explain that the site is down because we are currently under attack. Users need to remain calm, and the problem will be fixed as all the team is working on it (best efforts) and be polite.



! EXTRA

- HASHTAGS
An #experimentation and #POC for #plagiarism #checker #ScikitLearn #Python #Streamlit, https://bit.ly/3YI5Ubl

Using #ChatGPT on a #daily #work as a #PO, #Developer, for #QA or #Support and #checking #plagiarism if needed with #Python, https://wp.me/p3Vuhl-3dF

- GIT COMMIT
git status
git branch
git add .
git commit -am "updates files make it work for video"
git commit -am "change on several README files"
git push origin master



! VIDEOS

--- 001_ai_chatgpt_usages_po_qa_cypress.mov
+ https://youtu.be/dpQL2BK7wpk
- 001_ai_chatgpt_usages_po_qa_cypress.png
- 001_ai_chatgpt_usages_po_qa_cypress.mp4
#1 Using ChatGPT for P.O: Generate Jira tickets for user stories & some Q/A testing scripts (Cypress)
KW: ChatGPT, Cypress, P.O, Jira, mail, suport, ChatGPT, Developer, P.O, plagiarism, python, Q/A, support, usage, work

--- 002_ai_chatgpt_usages_python.mov
+ https://youtu.be/fjht2WMZlic
- 002_ai_chatgpt_usages_python.png
- 002_ai_chatgpt_usages_python_good.mov
#2 Using ChatGPT for coding Generate Python code with comments using 
Beautiful Soup & SQL command for SQLite


--- 003_ai_chatgpt_usages_php_wp_plugin.mov
+ https://youtu.be/drtNkrSgT8Q
- 003_ai_chatgpt_usages_php_wp_plugin.png
- 003_ai_chatgpt_usages_php_wp_plugin_good.mov
#3 Using ChatGPT to generate simple PHP  WordPress Plugin code to extend Dashboard WP features

--- 004_ai_chatgpt_usages_plagiarism_checker_python.mov
+ https://youtu.be/RTvFSqW4u2I
- 004_ai_chatgpt_usages_plagiarism_checker_python.png
- 004_ai_chatgpt_usages_plagiarism_checker_python_good.mov
#4 After using ChatGPT a Quick POC to check Plagiarism with Python


! MUSIC
you_re_not_that_funky_otis_mcdonald.mp3


! HASHTAGS

--- POST
An #experimentation and #POC for #plagiarism #checker #ScikitLearn #Python #Streamlit, https://bit.ly/3YI5Ubl

Using #ChatGPT on a #daily #work as a #PO, #Developer, for #QA or #Support and #checking #plagiarism if needed with #Python, https://wp.me/p3Vuhl-3dF

--- VIDEO
Much more attractive, 4 quick videos following this post, https://wp.me/p3Vuhl-3dF

+ https://youtu.be/dpQL2BK7wpk
Using #ChatGPT #POC #PO Generate #Jira #Tickets for #UserStories & some #QA #testing #scripts #Cypress, https://youtu.be/dpQL2BK7wpk

+ https://youtu.be/fjht2WMZlic
Using #ChatGPT #POC Generate #Python code with comments using 
#BeautifulSoup or #SQL command for #SQLite, https://youtu.be/drtNkrSgT8Q


+ https://youtu.be/drtNkrSgT8Q
Using #ChatGPT #POC generate #PHP #WordPress #Plugin Code to extend Dashboard #WP features, https://youtu.be/drtNkrSgT8Q



+ https://youtu.be/RTvFSqW4u2I
Using #ChatGPT #POC to #checking #plagiarism with #Python #streamlit #ScikitLearn, https://youtu.be/RTvFSqW4u2I

Using #ChatGPT #POC #PO #PHP #WordPress #Jira #Tickets #UserStories #Python #plagiarism #checker #ScikitLearn #Python #Streamlit. Code and Video Walkthrough for these videos can be found at https://bit.ly/3YI5Ubl


! FUN
Just for fun. #IA #MMA kind of fight #MammaMIA is reduced to #ProductivityNotDead #MakeYourJobEasier VS #AIDrivenDoom #WillRobotsTakeMyJob

Hard to find a clear position between #ProductivityNotDead #MakeYourJobEasier VS #AIDrivenDoom #WillRobotsTakeMyJob

Reminds also Bible Parable from "Il buono, il brutto, il cattivo". Dividing the World In Two. "You see, the world is divided into two categories. Those with a loaded gun and those who dig. You dig." Just replace #Gun by #AI.


! OTHER
- OUTRO
Thanks for watching.

This video is part of the post 
"Using ChatGPT on a daily work as a P.O, Developer or for Q/A or Support and checking plagiarism if needed with Python"

Check my blog @ http://www.flaven.fr
Check my Github @ http://github.com/bflaven/

Music
You're not that Funky
Otis McDonald
Royalty Free Music

Plus que jamais, prenez soin de vous et des vôtres,
Take care of yourself and the ones you loved.

© Flaven.fr February 2023
